### PR TITLE
chore(sdk): clean up targets in Makefile

### PR DESCRIPTION
**Pull request by dmitryduev**
Description
-----------
Added `make lint` to sort imports, format the code, and execute flake8 and mypy.

Testing
-------
Manually ran `make lint`.

Checklist
-------
- [x] Include reference to internal ticket "Fixes WB-NNNN" and/or GitHub issue "Fixes #NNNN" (if applicable)
- [x] Ensure PR title compliance with the [conventional commits standards](https://github.com/wandb/wandb/blob/master/CONTRIBUTING.md#conventional-commits)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -62,6 +62,9 @@ test-short:
 format:
 	tox -e format
 
+lint:
```
Comment (kptkin) [inline Makefile:65]: we have `test` in the makefile, what do you think about renaming that one?

**Review by kptkin — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (dmitryduev) [reply Makefile:65]: yea, perhaps removing it for good is a good idea. it's pretty misleading, to me at least

Comment (dmitryduev) [reply Makefile:62]: how about now?

File: Makefile [hunk 2]
```diff
@@ -51,17 +51,17 @@ test-clean:
...
 
 test:
-	tox -e "pyupgrade,black,mypy,flake8,docstrings"
-
-test-full:
 	tox
 
 test-short:
-	tox -e "pyupgrade,black,mypy,flake8,docstrings,py36"
+	tox -e "pyupgrade,isort-format,format,mypy,flake8,docstrings,py38"
```
Comment (raubitsj) [inline Makefile:57]: this should be isort-check and black right..... this is test, not change.

i also think of lint as just telling me something is wrong, not actually changing it.

I am all for renaming a bunch of these so it makes sense.. lets have a thread in slack and decide what makes sense

**Comment by dmitryduev** [comment]
closing as stale

**Pull request closed without merging**